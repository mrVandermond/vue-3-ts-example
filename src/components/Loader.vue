<template>
  <img
    class="loader"
    src="assets/loader.gif"
    alt="loader-image"
  >
</template>

<script lang="ts" setup>
</script>

<style lang="scss">
.loader {
  width: 48px;
  height: 48px;
}
</style>
