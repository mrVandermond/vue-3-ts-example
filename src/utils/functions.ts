export function getItemIndexBy<T>(array: T[], item: T, field: keyof T): number {
  return array.findIndex((temp) => temp[field] === item[field]);
}
