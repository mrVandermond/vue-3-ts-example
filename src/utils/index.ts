export * as utlFuncs from './functions';
export * as utlCst from './constants';
