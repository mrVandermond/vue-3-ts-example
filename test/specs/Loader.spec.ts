import Loader from '@/components/Loader.vue';

import type { TWrapperFactoryReturnType } from '../utils';
import { wrapperFactory } from '../utils';

describe('Loader component', () => {
  const getWrapper = (): TWrapperFactoryReturnType => wrapperFactory(Loader);

  it('Верстка', () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
