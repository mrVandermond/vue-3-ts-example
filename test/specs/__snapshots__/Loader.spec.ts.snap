// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Верстка 1`] = `
<img class="loader"
     src="assets/loader.gif"
     alt="loader-image"
>
`;
